### PR TITLE
PHPLIB-1367 Add enum for System variables

### PR DESCRIPTION
**Pull request by GromNaN**
Fix [PHPLIB-1367](https://jira.mongodb.org/browse/PHPLIB-1367)

Includes https://github.com/mongodb/mongo-php-builder/pull/63 (to be merged before)

I originally created factories for system variables, but we can also use enums. I'm not sure what's the best approach. They are almost the same for DX

### CODE REVIEW

**Review by alcaeus — APPROVED** [review]

File: src/Builder/Variable.php [hunk 1]
```diff
@@ -1,158 +0,0 @@
...
-        return new Expression\Variable('ROOT');
-    }
-
-    /**
-     * References the start of the field path being processed in the aggregation pipeline stage.
-     * Unless documented otherwise, all stages start with CURRENT the same as ROOT.
-     * CURRENT is modifiable. However, since $<field> is equivalent to $$CURRENT.<field>, rebinding
-     * CURRENT changes the meaning of $ accesses.
-     */
-    public static function current(string $fieldPath = ''): ResolvesToAny
```
Comment (GromNaN) [inline src/Builder/Variable.php:61]: This helper was more explicit than the new `Variable::Current->dot(...)` function in introduced to the enum.

File: tests/Builder/Expression/UnsetFieldOperatorTest.php [hunk 2]
```diff
@@ -38,7 +39,7 @@ public function testRemoveFieldsThatContainPeriods(): void
             Stage::replaceWith(
                 Expression::unsetField(
                     field: 'price.usd',
-                    input: Expression::variable('ROOT'),
+                    input: Variable::Root,
```
Comment (GromNaN) [inline tests/Builder/Expression/UnsetFieldOperatorTest.php:42]: This could have been `Variable::root()`, which is as good as the enum `Variable::Root`.

**Review by GromNaN — COMMENTED** [review]
I'm still not convinced this approach is better than the factory methods.

**Comment by GromNaN** [comment]
Closing for now. Lets revisit later. Ticket still open.

**Pull request closed without merging**